### ISSUE TITLE
File-level ignore `reportUnnecessaryTypeIgnoreComment` not honored for EOL comments

### ISSUE THREAD

**Issue by felixm-stripe** [microsoft/pyright#8223]
Having EOL type-ignores after an ignore is intended behavior as per #4294. However, should a file-level directive take precendence here?

Given:

```python
# type: ignore[reportUnnecessaryTypeIgnoreComment]

# import json # type: ignore
```

yields the output:

```
/path/to/foo.py:3:17 - error: Unnecessary "# type: ignore" comment
1 error, 0 warnings, 0 informations
```

when invoked with Pyright CLI using version 1.1.351.

#### Considerations
When an engineer comments out portions of source code in order to iterate, Pyright still complains without a supported escape-hatch.

#### Workarounds
Currently, our engineers work around this by using multiline comments to work around #4294.

**Comment by erictraut** [comment]
This is working as intended.

A line-level directive always overrides a file-level directive, and a file-level directive always overrides a project-level directive.

Diagnostic-specific directives are not supported for `# type: ignore`. They are supported only for `# pyright: ignore`.

If you want to disable a specific diagnostic for an entire file, you should use the comment `# pyright: reportUnnecessaryTypeIgnoreComment=false` at the top of the file. See [this documentation](https://microsoft.github.io/pyright/#/comments?id=file-level-type-controls) for details.

**Comment by felixm-stripe** [comment]
That escape hatch is fine! Thanks (=
